src/Microsoft.Performance.SDK/Processing/ColumnBuilding/IColumnBuilder.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Performance.SDK.Processing.ColumnBuilding;
+
+/// <summary>
+///     Base interface for classes that can build/extend upon columns that are
+///     part of a table.
+/// </summary>
+public interface IColumnBuilder
+{
+    /// <summary>
+    ///     Builds the column. Note that this method MUST be called to have
+    ///     any effect on the column being built. Calling this method multiple
+    ///     times will result in only the final call having any effect.
+    /// </summary>
+    void Build();
```

Comment:
>Build(); [](http://example.com/codeflow?start=9&length=8)

Opinion:

When I first read through ColumnVariantsTests.BuildingTwiceOverridesFirstBuild(), I was scratching my head trying to understand what was going on.

I believe the issue for me is the naming of this method is non-intuitive.

Sure, it builds the variant, but that part is mostly meaningless because, if it's not *applied* to the processor, then it does nothing - and that's the key to this method.

I believe that changing the name to Apply() or Process() or similar would make using this more intuitive.

Reply:
I can see that confusion. `Build` usually returns an object being built.

I am OK with `Apply()`. However, I think `Commit()` might also work. Do you prefer one over the other?

Reply:
Hrm, I think that I like Commit

Reply:
Changed to Commit